pkg/virtualnode/resource_manager.go

```diff
@@ -133,8 +134,33 @@ func (m *defaultResourceManager) validateMeshDependencies(ctx context.Context, m
 
 // findVirtualServiceDependencies find the VirtualService dependencies for this virtualNode.
 func (m *defaultResourceManager) findVirtualServiceDependencies(ctx context.Context, vn *appmesh.VirtualNode) (map[types.NamespacedName]*appmesh.VirtualService, error) {
-	vsByKey := make(map[types.NamespacedName]*appmesh.VirtualService, len(vn.Spec.Backends))
+	vsByKey := make(map[types.NamespacedName]*appmesh.VirtualService)
 	vsRefs := ExtractVirtualServiceReferences(vn)
+	for _, backendGroupRef := range vn.Spec.BackendGroups {
+		// Wildcard special case
+		bgKey := references.ObjectKeyForBackendGroupReference(vn, backendGroupRef)
+		if bgKey.Name == "*" {
+			var listOptions client.ListOptions
+			listOptions.Namespace = bgKey.Namespace
+			vsList := &appmesh.VirtualServiceList{}
+			if err := m.k8sClient.List(ctx, vsList, &listOptions); err != nil {
+				return nil, fmt.Errorf("could not list virtualservices for namespace: %s", bgKey.Namespace)
+			}
+			for _, vs := range vsList.Items {
+				vsRef := appmesh.VirtualServiceReference{
+					Namespace: aws.String(vs.Namespace),
+					Name:      vs.Name,
+				}
+				vsRefs = append(vsRefs, vsRef)
+			}
+		} else {
+			bg, err := m.referencesResolver.ResolveBackendGroupReference(ctx, vn, backendGroupRef)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to resolve backendGroupRef")
+			}
+			vsRefs = append(vsRefs, bg.Spec.VirtualServices...)
+		}
+	}
 	for _, vsRef := range vsRefs {
 		vsKey := references.ObjectKeyForVirtualServiceReference(vn, vsRef)
 		if _, ok := vsByKey[vsKey]; ok {
```

Comment:
this is nice that the existing logic does a map here since we dont have to dedupe virtual services if customers configure overlapping backendgroups